rasa/core/actions/forms.py

```diff
@@ -520,7 +523,54 @@ async def validate(
                 self.name(),
                 f"Failed to extract slot {slot_to_fill} with action {self.name()}",
             )
-        return validation_events
+
+        # Add slot_filled_action
+        slot_filled_action_events = []
+
+        # Create temporary tracker with the validation events applied
+        # Otherwise, the slots will not be set
+        temp_tracker = self._temporary_tracker(
+            tracker, validation_events, domain
+        )
+
+        slots_filled = [
+            event.key
+            for event in validation_events
+            if isinstance(event, SlotSet) and not event.key == REQUESTED_SLOT
+        ]
+
+        for slot_filled in slots_filled:
```

Comment:
The order is governed by the validation events. Perhaps better to use the order of the slots defined in the form.